### DISCUSSION TITLE
Gas estimation failed : require function

### DISCUSSION THREAD

**Discussion by AyuK03** [smartcontractkit/full-blockchain-solidity-course-py#1737] [Q&A]
I coded as told in the course for the fund_me contract and even used the exact same wei amount in the value input during deployment. But I'm still getting the Gas estimation failed message though the value I put was way higher than the minimum amount put in require func.

I typed :
`require(getConversionRate(msg.value)>= minimumUSD, "You need to spend more ETH !");`

For the value in deployment funcn :
100000000000000000 Wei

Its showing ;
`Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending?
err: insufficient funds for gas * price + value: address 0x664163eB7CC5c24c503122479960d7f70b963bA9 have 95294835567893581 want 100000000000000000 (supplied gas 15010499)

Kindly help me through
`

**Comment by cromewar** [comment]
Hello @AyuK03 
The easiest way to solve this is just reduce the amount you are sending so your wallet has enough funds to send over the transaction.
If you are testing on local ganache, you shouldn't have problems as the accounts have a lot of eth.
So please in order to review this, please share the code here in order we can check it.

**Reply by AyuK03** [reply]
``` Solidity
// SPDX-License-Identifier: MIT

pragma solidity >=0.8.9 <0.9.0;

import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol";

contract FundMe {

    using SafeMathChainlink for uint256;

    mapping (address => uint256) public addressToAmountFunded;

    function fund() public payable {

        uint256 minimumUSD = 50 * 10 ** 18;

        require(getConversionRate(msg.value)>= minimumUSD, "You need to spend more ETH !");
        
        addressToAmountFunded[msg.sender] += msg.value;

        // must know what the ETH -> USD conversion rate is

    
    }

    function getVersion() public view returns(uint256){
        AggregatorV3Interface priceFeed_latest = AggregatorV3Interface(0xD4a33860578De61DBAbDc8BFdb98FD742fA7028e);
    
        return priceFeed_latest.version();
    }

    function getPrice() public view returns(uint256){
        AggregatorV3Interface price_feed = AggregatorV3Interface(0xD4a33860578De61DBAbDc8BFdb98FD742fA7028e);
        
        /*(uint80 roundId,
         int256 answer,
         uint256 startedAt,
         uint256 updatedAt,
         uint80 answeredInRound)
         = price_feed.latestRoundData();
         return uint256(answer);
        */

        (,int256 answer,,,) = price_feed.latestRoundData();
        return uint256(answer * 10000000000); // will get price of eth in USD
    }

    function getConversionRate(uint256 ethAmount) public view returns(uint256) {
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount)/ 1000000000000000000;
        return ethAmountInUsd;
    }
    
}
```

> The value that I had put for transaction in Injected Provider- Metamask was just around 1000-10000 gwei & I have 0.1792 GoerliETH in my local testnet

**Reply by cromewar** [reply]
on the meantime I reproduce this could you change the minimum from $50 to $5 and test it again?
